src/hotspot/share/gc/shenandoah/shenandoahHeap.cpp

```diff
@@ -1618,7 +1622,7 @@ void ShenandoahHeap::op_init_mark() {
   // Make above changes visible to worker threads
   OrderAccess::fence();
 
-  concurrent_mark()->mark_roots(ShenandoahPhaseTimings::scan_roots);
+  generation->concurrent_mark()->mark_roots(ShenandoahPhaseTimings::scan_roots);
```

Comment:
I am wondering if it makes sense to have an instance field / an API to get the "active generation". It may reduce API changes.

Reply:
We have that in later commits and we will use it in other phases, just not in marking. We aspire to having two marking phases running concurrently with each other, later on, past milestone 6.